packages/node-experimental/src/integrations/http.ts

```diff
@@ -127,6 +129,11 @@ export class Http implements Integration {
           requireParentforIncomingSpans: false,
           requestHook: (span, req) => {
             this._updateSpan(span, req);
+
+            // Update the isolation scope, isolation this request
+            if (getSpanKind(span) === SpanKind.SERVER) {
+              setIsolationScope(Scope.clone(getIsolationScope()));
```

Comment:
This is the actual code for auto-isolation with OTEL. Should work for any instrumentation that builds on top of http, so e.g. both express and fastify etc. are covered by this. (in theory, more testing will be needed)